lib/signaling/v2/room.js

```diff
@@ -719,6 +749,76 @@ function handleSubscriptions(room) {
   });
 }
 
+/**
+ * NOTE(mmalavalli): Since A/V sync metrics are not part of the public StatsReport class, we add them
+ * only for reporting purposes.
+ * @private
+ */
+function addAVSyncMetricsToLocalTrackStats(trackStats, trackResponse, movingAverageDeltas) {
+  const {
+    framesEncoded,
+    packetsSent,
+    totalEncodeTime,
+    totalPacketSendDelay
+  } = trackResponse;
+  const augmentedTrackStats = Object.assign({}, trackStats);
+  const trackMovingAverageDeltas = movingAverageDeltas.get(trackStats.trackSid) || new Map();
+
+  if (typeof totalEncodeTime === 'number' && typeof framesEncoded === 'number') {
+    const trackAvgEncodeDelayMovingAverageDelta = trackMovingAverageDeltas.get('avgEncodeDelay')
+      || new MovingAverageDelta(3);
+    trackAvgEncodeDelayMovingAverageDelta.putSample(totalEncodeTime * 1000, framesEncoded);
+    augmentedTrackStats.avgEncodeDelay = trackAvgEncodeDelayMovingAverageDelta.get();
+    trackMovingAverageDeltas.set('avgEncodeDelay', trackAvgEncodeDelayMovingAverageDelta);
+  }
+  if (typeof totalPacketSendDelay === 'number' && typeof packetsSent === 'number') {
+    const trackAvgPacketSendDelayMovingAverageDelta = trackMovingAverageDeltas.get('avgPacketSendDelay')
+      || new MovingAverageDelta(3);
+    trackAvgPacketSendDelayMovingAverageDelta.putSample(totalPacketSendDelay * 1000, packetsSent);
+    augmentedTrackStats.avgPacketSendDelay = trackAvgPacketSendDelayMovingAverageDelta.get();
+    trackMovingAverageDeltas.set('avgPacketSendDelay', trackAvgPacketSendDelayMovingAverageDelta);
+  }
+  movingAverageDeltas.set(trackStats.trackSid, trackMovingAverageDeltas);
+  return augmentedTrackStats;
+}
+
+/**
+ * NOTE(mmalavalli): Since A/V sync metrics are not part of the public StatsReport class, we add them
+ * only for reporting purposes.
+ * @private
+ */
+function addAVSyncMetricsToRemoteTrackStats(trackStats, trackResponse, movingAverageDeltas) {
+  const {
+    estimatedPlayoutTimestamp,
+    framesDecoded,
+    jitterBufferDelay,
+    jitterBufferEmittedCount,
+    totalDecodeTime
+  } = trackResponse;
+  const augmentedTrackStats = Object.assign({}, trackStats);
+  const trackMovingAverageDeltas = movingAverageDeltas.get(trackStats.trackSid) || new Map();
```

Comment:
with simulcast enabled would you  get multiple tracks with same trackSid and different `ssrc` - Shouldn't they be kept separately?